File: go/04_loops.go
package main

import "fmt"

func main() {
	// for as classic loop
	for i := 0; i < 5; i++ {
		fmt.Println(i)
	}

	// for as while
	count := 0
	for count < 3 {
		fmt.Println("count is", count)
		count++
	}

	// range over slice
	fruits := []string{"apple", "banana", "cherry"}
	for _, fruit := range fruits {
		fmt.Println(fruit)
	}
}

File: cpp/04_loops.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main() {
    for (int i = 0; i < 5; i++) {
        cout << i << endl;
    }

    vector<string> fruits = {"apple", "banana", "cherry"};
    for (const string& fruit : fruits) {
        cout << fruit << endl;
    }

    int count = 0;
    while (count < 3) {
        cout << "count is " << count << endl;
        count++;
    }

    return 0;
}
